Assets/Script/UI/EnemyHealthBar.cs

```diff
@@ -53,13 +53,23 @@ private void LateUpdate()
         canvasGroup.alpha = alphaToUse;
     }
 
-    public void EnemyStatsChangeCallback(float newHitPoint)
+    public void EnemyStatsChangeCallback(EnemyStats newStats)
     {
-        canvas.SetActive(true);
-        onTimeAlpha = 1.0f;
-        decreasingOnTimeAlpha = false;
-        CancelInvoke();
-        Invoke("disableVisualize", onTime);
+        var newHitPoint = newStats.CurrentHitPoint;
+        maxHitPoint = newStats.MaxHitPoint;
+
+        if(_isFirstSetup)
+        {
+            _isFirstSetup = false;
+        }
+        else
+        {
+            canvas.SetActive(true);
+            onTimeAlpha = 1.0f;
+            decreasingOnTimeAlpha = false;
+            CancelInvoke();
+            Invoke("disableVisualize", onTime);
+        }
 
         healthBarImage.color = colorGradient.Evaluate(newHitPoint / maxHitPoint);
```

Comment:
healthBarSlider.maxValue = maxHitPoint;
Add this before setting the value

Reply:
Added